Core/Core/Logging/SpeckleException.cs

```diff
@@ -1,36 +1,37 @@
 ﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+using GraphQL;
+using Sentry.Protocol;
+
 namespace Speckle.Core.Logging
 {
   public class SpeckleException : Exception
   {
+    public List<KeyValuePair<string, object>> GraphQLErrors { get; set; }
     public SpeckleException()
     {
     }
 
-    public SpeckleException(string message) : base(message)
-    {
-    }
-
-    public SpeckleException(string message, Exception inner)
-         : base(message, inner)
-    {
-    }
-  }
-
-
-  public class GraphQLException : Exception
-  {
-    public GraphQLException()
+    public SpeckleException(string message, bool log = false, SentryLevel level = SentryLevel.Info ) : base(message)
     {
+      if (log)
+        Log.CaptureException(this, level);
     }
 
-    public GraphQLException(string message) : base(message)
+    public SpeckleException(string message, GraphQLError[ ] errors, bool log = false,
+      SentryLevel level = SentryLevel.Info) : base(message)
     {
+      GraphQLErrors = errors.Select(error => new KeyValuePair<string, object>("error", error.Message)).ToList();
+      if (log)
+        Log.CaptureException(this, extra: GraphQLErrors);
     }
 
-    public GraphQLException(string message, Exception inner)
+    public SpeckleException(string message, Exception inner, bool log = false, SentryLevel level = SentryLevel.Info)
```

Comment:
We should probably check if `inner is SpeckleException` and in that case ignore it as it's already been reported before.